### DISCUSSION TITLE
Fabricjs objects disappear after animation is done in material stepper in angular

### DISCUSSION THREAD

**Discussion by zmiju** [fabricjs/fabric.js#8854] [Q&A]
I added fabricjs canvas in one step of material stepper. It is presented pretty well only while animating entrance of this step. But when animation is finished all objects suddenly disappear and canvas looks empty.

I tried to delay presenting the canvas or lazy loading the step of the material stepper. But it didn't help.

When I increased animation duration to 10 seconds, I can clearly see that it is related with completion of the animation. When I completely disable the animation the canvas is empty as well.

Here is the code: There is parent component:

```
<mat-horizontal-stepper [linear]="true" #stepper animationDuration="10000" (animationDone)="stepperDone()">
<mat-step>
       <app-dotmap [processKey]="'process0'" ></app-dotmap>
</mat-step>
</mat-horizontal-stepper>
And child componentn app-dotmap:
```

`<canvas id="{{processKey}}" style="border: 1px solid black;"></canvas>`

And its typescript:

```
canvas: any
Input()processKey = ""
  initiateCanvas() {
    this.canvas = new fabric.Canvas(this.processKey, {
      selection: false,
      centeredScaling:true,
      controlsAboveOverlay: true,
      targetFindTolerance: 4
    });
    let dot = new fabric.Circle()
    dot.left = left
    dot.top = top
    dot.radius = 20
    dot.fill = "blue"
    this.canvas.add(dot)
}
```

Just after animation is done, blue dot on canvas disappears.

Help please. I don't know if it is me, fabric or angular.

**Comment by ShaMan123** [comment]
No repro = no attention
Don't expect anyone to waste time trying to work on it

**Reply by zmiju** [reply]
I have managed to solve it, thanks.
